scripts/globals/strangeapparatus.lua

```diff
@@ -28,62 +29,227 @@ local DARK_CLUSTER      = 4111
-    [194] = {'HR', 5, PURPLE_CHIP, LIGHTNING_CLUSTER, 17572203}, -- Outer Horutoto Ruins
-    [195] = {'EN', 4, CLEAR_CHIP,  ICE_CLUSTER,       17576271}, -- The Eldieme Necropolis
-    [196] = {'GM', 1, YELLOW_CHIP, EARTH_CLUSTER,     17580340}, -- Gusgen Mines
-    [197] = {'CN', 2, BLUE_CHIP,   WATER_CLUSTER,     17584427}, -- Crawler's Nest
-    [198] = {'MS', 7, BLACK_CHIP,  DARK_CLUSTER,      17588704}, -- Maze of Shakhrami
-    [200] = {'GC', 6, WHITE_CHIP,  LIGHT_CLUSTER,     17596729}, -- Garlaige Citadel
-}
-
-------------------------------------
--- Strange Apparatus rewards
--- [zoneId] = {itemid, drop rate, drop quantity, itemid, drop rate, drop quantity, ...}
--- Must be in ascending order by drop rate
-------------------------------------
-
-local strAppDrop =
-{
-    [191] = {17093, 0.0400, 1, 17461, 0.0800, 1, 18084, 0.1200, 1, 17158, 0.1600, 1, 16563, 0.2000, 1, 12742, 0.2400, 1, 16647, 0.2800, 1, 18206, 0.3200, 1, 16408, 0.3600, 1, 221, 0.4000, 1, 17333, 0.5200, 6, 1229, 0.7000, 2, 931, 1.0000, 8},
-    [193] = {17093, 0.0400, 1, 17461, 0.0800, 1, 18084, 0.1200, 1, 17158, 0.1600, 1, 16563, 0.2000, 1, 12742, 0.2400, 1, 16647, 0.2800, 1, 18206, 0.3200, 1, 16408, 0.3600, 1, 221, 0.4000, 1, 17333, 0.5200, 6, 1229, 0.7000, 2, 931, 1.0000, 8},
-    [194] = {17093, 0.0400, 1, 17461, 0.0800, 1, 18084, 0.1200, 1, 17158, 0.1600, 1, 16563, 0.2000, 1, 12742, 0.2400, 1, 16647, 0.2800, 1, 18206, 0.3200, 1, 16408, 0.3600, 1, 221, 0.4000, 1, 17333, 0.5200, 6, 1229, 0.7000, 2, 931, 1.0000, 8},
-    [195] = {17093, 0.0400, 1, 17461, 0.0800, 1, 18084, 0.1200, 1, 17158, 0.1600, 1, 16563, 0.2000, 1, 12742, 0.2400, 1, 16647, 0.2800, 1, 18206, 0.3200, 1, 16408, 0.3600, 1, 221, 0.4000, 1, 17333, 0.5200, 6, 1229, 0.7000, 2, 931, 1.0000, 8},
-    [196] = {17093, 0.0400, 1, 17461, 0.0800, 1, 18084, 0.1200, 1, 17158, 0.1600, 1, 16563, 0.2000, 1, 12742, 0.2400, 1, 16647, 0.2800, 1, 18206, 0.3200, 1, 16408, 0.3600, 1, 221, 0.4000, 1, 17333, 0.5200, 6, 1229, 0.7000, 2, 931, 1.0000, 8},
-    [197] = {17093, 0.0400, 1, 17461, 0.0800, 1, 18084, 0.1200, 1, 17158, 0.1600, 1, 16563, 0.2000, 1, 12742, 0.2400, 1, 16647, 0.2800, 1, 18206, 0.3200, 1, 16408, 0.3600, 1, 221, 0.4000, 1, 17333, 0.5200, 6, 1229, 0.7000, 2, 931, 1.0000, 8},
-    [198] = {17093, 0.0400, 1, 17461, 0.0800, 1, 18084, 0.1200, 1, 17158, 0.1600, 1, 16563, 0.2000, 1, 12742, 0.2400, 1, 16647, 0.2800, 1, 18206, 0.3200, 1, 16408, 0.3600, 1, 221, 0.4000, 1, 17333, 0.5200, 6, 1229, 0.7000, 2, 931, 1.0000, 8},
-    [200] = {17093, 0.0400, 1, 17461, 0.0800, 1, 18084, 0.1200, 1, 17158, 0.1600, 1, 16563, 0.2000, 1, 12742, 0.2400, 1, 16647, 0.2800, 1, 18206, 0.3200, 1, 16408, 0.3600, 1, 221, 0.4000, 1, 17333, 0.5200, 6, 1229, 0.7000, 2, 931, 1.0000, 8},
+    [dsp.zone.DANGRUF_WADI] =
+    {
+        suffix  = 'DW',
+        uid     = 0,
+        chip    = RED_CHIP,
+        cluster = FIRE_CLUSTER,
+        drop    =
+        {
+            17093, 0.0400, 1,
```

Comment:
I can has comments saying which drop is what? to save future contributors having to run to item_basic/ffxiah

code looks good. I'm assuming you tested.

Reply:
Added item names in comments, and some TODOs.  The drop data that's in there now is identical per zone.  Wiki indicates that certain items are more likely to drop in certain zones.  Also it seems that three new items were added to the drop pools in 2017 (locations and rates unknown).

I tested before PRing.